### PR TITLE
Improve handling of expression in metadata

### PR DESCRIPTION
**Pull request by ven-k**
Instead of evaluating expression while defining the model, keep it as is and execute while creating an instance of it.

This now works:
```julia
@mtkmodel Model begin
    @parameters begin
        ρ₀
        p′
        β
    end
    @variables begin
        r(t), [guess = ρ₀*(1 + p′/β)]
    end
end
```

## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API

### CODE REVIEW

**Comment by ven-k** [comment]
New tests are added to "Model Parsing" and they pass [here](https://github.com/SciML/ModelingToolkit.jl/actions/runs/8832467051/job/24249893731?pr=2501#step:6:1301).

Test failures are unrelated to this PR.

Documentation had passed for the first commit [here](https://github.com/SciML/ModelingToolkit.jl/actions/runs/8830257057/job/24242990387?pr=2501).
The current failure is CI related and not this PR.

**Comment by ven-k** [comment]
This PR is ready for review.

**Review by ChrisRackauckas — APPROVED** [review]

File: mo.diff [hunk 1]
```diff
@@ -0,0 +1,582 @@
+27,28d26
```
Comment (sjkelly) [inline mo.diff:1]: This shouldn't have been added?

Comment (ChrisRackauckas) [reply mo.diff:1]: yes

Comment (ven-k) [reply mo.diff:1]: I've removed the file with this commit https://github.com/SciML/ModelingToolkit.jl/pull/2583/commits/27e26992589337f5d6333983a91ccc474d34973d

**Pull request merged**